test/BodyParams/BodyParamsMiddlewareTest.php

```diff
@@ -181,13 +188,43 @@ public function testThrowsMalformedRequestBodyExceptionWhenRequestBodyIsNotValid
         $this->expectExceptionMessage($expectedException->getMessage());
         $this->expectExceptionCode($expectedException->getCode());
 
-        $middleware($serverRequest, new Response(),
-            function ($request, $response) use (&$triggered) {
-                $triggered = true;
-                return $response;
-            }
+        $middleware->process(
+            $serverRequest,
+            $this->fakeDelegateNeverTriggered(
+                function (ServerRequestInterface $request) use (&$triggered) {
+                    $triggered = true;
+
+                    return new Response();
+                }
+            )
         );
 
         $this->assertFalse($triggered, 'Next should not have been triggered!');
     }
+
+    private function fakeDelegate(callable $callback)
+    {
+        $delegate = $this->createMock(DelegateInterface::class);
```

Comment:
Yes, that was very interesting actually.  I tried this first:

```php
$serverRequest = new ServerRequest([], [], '', 'PUT', $this->body, ['Content-type' => $contentType]);

$delegate = $this->prophesize(DelegateInterface::class);
$delegate->process(Argument::type(RequestInterface::class))->will(function ($req) use (&$serverRequest) {
    $serverRequest = $req; // <----- $req is a ServerRequest wrapped in an array

    return new Response();
});

$this->bodyParams->process(
    $serverRequest,
    $delegate->reveal()
);

$this->assertSame(
    json_encode(['foo' => 'bar']),
    $serverRequest->getAttribute('rawBody')
);
$this->assertSame(['foo' => 'bar'], $serverRequest->getParsedBody());

// Fails with:
//
// ZendTest\Expressive\Helper\BodyParams\BodyParamsMiddlewareTest::testParsesRawBodyAndPreservesRawBodyInRequestAttribute
// with data set #0 ('application/json')
// Error: Call to a member function getAttribute() on array
```

For whatever reason prophesize injected the ServerRequest wrapped in an array. I don't know if that's by design, a bug or something I did wrong. But since I've use the PHPUnit built-ins without problems somewhere else, I used those instead.